thunder/torch/__init__.py

```diff
@@ -1102,6 +1102,11 @@ def ceil(a):
     return clang.ceil(a)
 
 
+@torchsymbol(torch.is_complex, is_method=True)
```

Comment:
Let's not make this a symbol and let's put it next to `is_cuda`:

https://github.com/khushi-411/lightning-thunder/blob/05524ae8b828640587f6a67c72cd0c78671df649/thunder/torch/__init__.py#L191

But then we need to two things (I think) to make it work properly:

- call `register_method()`
- update the map `_torch_to_thunder_function_map[torch.is_complex] = is_complex`

Then I would expect the function and method to work as expected

Reply:
Thanks for taking this on, @khushi-411! Mike had me do the same in #429, if you want more detail here.

Note that if it's both a method and a function of the module root (i.e. `torch.is_complex`) then AFAICT one needs to register both `torch.is_complex` and `torch.Tensor.is_complex`.

Reply:
Hi, @tfogal! Thanks very much for the heads-up. I'll look into the PR and keep you posted.

I'm currently travelling for a week, but I will complete this PR soon. I apologize for the delay.